File: Data Structures & Algorithms/max-water-container/submission-3.java
class Solution {
    public int maxArea(int[] heights) {

        int l = 0;
        int r = heights.length - 1;
        int maxArea = Integer.MIN_VALUE;

        while(l < r)
        {
            int y = Math.min(heights[l], heights[r]);
            int x = r-l;
            maxArea = Math.max(maxArea, x*y);

            if(heights[l] > heights[r])
            {
                r--;
            }
            else
            {
                l++;
            } 

        }

        return maxArea;
        
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-3.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        # move left right on k since that needs to be minimized
        # piles/k sum < h
        # no need to sort piles running on k
        l = 1
        r = max(piles)

        print(r)

        min_k = max(piles)

        while l <= r:
            time = 0
            rate = (l+r)//2 #speed
            print(rate)
            for bananas in piles:
                time = time + math.ceil(bananas/rate)
            print("TIME" ,time)

            if time > h:
                l = rate+1
                print("L" ,l)
            else:
                min_k = min(min_k, rate)
                r = rate-1
                print("r" ,r)

        return min_k


            


        